localstack/services/events/scheduler.py

```diff
@@ -58,12 +98,7 @@ def disable_job(self, job_id):
                 break
 
     def cancel_job(self, job_id):
-        i = 0
-        while i < len(self.jobs):
-            if self.jobs[i].job_id == job_id:
-                del self.jobs[i]
-            else:
-                i += 1
+        self.jobs = [job for job in self.jobs if job.job_id != job_id]
```

Comment:
cleaner

Reply:
👍 
💡 We still need to make such modifications thread-safe using locks in the scheduler rework.